index.js

```diff
@@ -62,17 +62,15 @@ module.exports = function(options) {
 
     let origin;
     if (typeof options.origin === 'function') {
-      origin = options.origin(ctx);
-      if (origin instanceof Promise) origin = await origin;
+      origin = await Promise.resolve().then(() => options.origin(ctx));
```

Comment:
origin could be a normal function, why we need to change it to a promise?

Reply:
I think you're asking "why would we use `Promise.resolve().then(...)`" instead of just using `await options.origin(ctx)`. I updated to use that simpler syntax.

If that's not what you're asking, could you elaborate a little more on your question?